frontend/dashboard/pages/PageLayout/DashboardHeader/DashboardHeader.test.tsx

```diff
@@ -202,6 +211,78 @@ describe('DashboardHeader', () => {
     expect(mockNavigate).toHaveBeenCalledTimes(1);
   });
 
+  it('should not render the submenu when showSubMenu is false', () => {
+    (useParams as jest.Mock).mockReturnValue({
+      selectedContext: mockOrgTtd,
+      subroute: Subroute.OrgLibrary,
+    });
+    (useLocation as jest.Mock).mockReturnValue({ pathname: mockPathnameOrgLibraryTtd });
+
+    renderDashboardHeader({ showSubMenu: false, isRepoError: false });
+
+    expect(getFetchChangesButton()).not.toBeInTheDocument();
+  });
+
+  it('should not render the submenu when isRepoError is true', () => {
+    (useParams as jest.Mock).mockReturnValue({
+      selectedContext: mockOrgTtd,
+      subroute: Subroute.OrgLibrary,
+    });
+    (useLocation as jest.Mock).mockReturnValue({ pathname: mockPathnameOrgLibraryTtd });
+
+    renderDashboardHeader({ showSubMenu: true, isRepoError: true });
+
+    expect(getFetchChangesButton()).not.toBeInTheDocument();
+  });
+
+  it('should not render the submenu when the page is not orgLibrary', () => {
+    (useParams as jest.Mock).mockReturnValue({
+      selectedContext: mockOrgTtd,
+      subroute: Subroute.AppDashboard,
+    });
+    (useLocation as jest.Mock).mockReturnValue({ pathname: mockPathnameOrgLibraryTtd });
+
+    renderDashboardHeader({ showSubMenu: true, isRepoError: false });
+
+    expect(getFetchChangesButton()).not.toBeInTheDocument();
+  });
+
+  it('should not render the submenu when page is orgLibrary and feature flag is not on', () => {
+    (useParams as jest.Mock).mockReturnValue({
+      selectedContext: mockOrgTtd,
+      subroute: Subroute.OrgLibrary,
+    });
+    renderDashboardHeader({ showSubMenu: true, isRepoError: false });
```

Comment:
I think the location mock is missing here - without that the SubHeader will not render, no matter what the feature flag status is.
```suggestion
    (useLocation as jest.Mock).mockReturnValue({ pathname: mockPathnameOrgLibraryTtd });
    renderDashboardHeader({ showSubMenu: true, isRepoError: false });
```